### PR TITLE
Add missing urls for array docs

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
r? @frewsxcv

### CODE REVIEW

File: src/libstd/primitive_docs.rs [hunk 1]
```diff
@@ -347,13 +330,31 @@ mod prim_pointer { }
 /// ```
 ///
 /// If the array has 32 or fewer elements (see above), you can also use the
-/// array reference's `IntoIterator` implementation:
+/// array reference's [`IntoIterator`] implementation:
```
Comment (frewsxcv) [inline src/libstd/primitive_docs.rs]: This could be:

```
[`IntoIterator`][intoiterator]
```

**Review by frewsxcv — APPROVED** [review]
r=me with my comment addressed

File: src/libstd/primitive_docs.rs (hunk 1 above)
Comment (GuillaumeGomez) [inline src/libstd/primitive_docs.rs]: We generally try to keep the syntax I've put in here. So not sure if I should change it or not...

Comment (frewsxcv) [inline src/libstd/primitive_docs.rs]: Want to change the item using [intoiterator] then?

Comment (GuillaumeGomez) [inline src/libstd/primitive_docs.rs]: I don't intend to, however I can "reverse" the other urls if you prefer.

**Pull request merged**